lib/rubocop/cop/style/return_nil.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop enforces consistency between 'return nil' and 'return'.
+      #
+      # Supported styles are: return, return_nil.
+      #
+      # @example
+      #   # bad if EnforcedStyle is return, good if EnforcedStyle is return_nil
+      #   def abcd
+      #     return nil
+      #   end
+      #
+      #   # bad if EnforcedStyle is return_nil, good if EnforcedStyle is return
+      #   def abcd
+      #     return
+      #   end
+      class ReturnNil < Cop
+        include ConfigurableEnforcedStyle
+
+        RETURN_MSG = 'Use `return` instead of `return nil`.'.freeze
+        RETURN_NIL_MSG = 'Use `return nil` instead of `return`.'.freeze
+
+        def_node_matcher :return_nil_node?, '(return nil)'
+        def_node_matcher :return_node?, '(return)'
+
+        def on_return(node)
+          add_offense(node) unless correct_style?(node)
+        end
+
+        private
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrected = style == :return ? 'return' : 'return nil'
+            corrector.replace(node.source_range, corrected)
+          end
+        end
+
+        def message(_node)
+          style == :return ? RETURN_MSG : RETURN_NIL_MSG
+        end
+
+        def correct_style?(node)
+          ((style == :return) && !return_nil_node?(node)) ||
```

Comment:
Not sure why the negation is here? I find `style == :return && return_node?(node)` quite straightforward. 🙂 

Also, all the parens are redundant. 😉